packages/ns-storage/files/30_ns-storage

```diff
@@ -1,9 +1,28 @@
 #!/bin/sh
 
-[ "$(uci -q get rsyslog.ns_memory)" == "selector" ] && exit 0
 
-echo '# Rotate in-memory log if size is bigger than 50MB' >> /etc/rsyslog.conf
-echo '$outchannel log_rotation,/var/log/messages, 52428800, /usr/sbin/rotate-messages' >> /etc/rsyslog.conf
+# Check if rsyslog.ns_log is set to 'log_rotator', if so, exit
+[ "$(uci -q get rsyslog.ns_log)" = "log_rotator" ] && exit 0
+
+# Default log_size in case of error
+log_size=52428800
+
+# Try to extract tmpfs size, fall back to default log_size if an error occurs
+tmpfs_size=$(df -k /tmp | awk 'NR==2 {print $2}')
+if [ -z "$tmpfs_size" ] || ! echo "$tmpfs_size" | grep -qE '^[0-9]+$'; then
+	echo "Error: Could not retrieve tmpfs size, defaulting log_size to 52428800" >&2
+else
+	# Calculate 10% of tmpfs size in bytes
+	ten_percent_bytes=$(( tmpfs_size * 1024 / 10 ))
+
+	# Use default log_size if the calculated value is less than 52428800
+	if [ "$ten_percent_bytes" -gt "$log_size" ]; then
+		log_size=$ten_percent_bytes
+	fi
+fi
```

Comment:
I really like the logic of the minimum log_size :+1: 
Still, the current implementation has a big caveats: the `30_ns-storage` uci-default is ignored during package updates and it's not present after install; an uci-default can be used only during an image upgrade.

So this is my proposal:
- keep line 3 unchanged and remove lines from 4 to 6 (see the comment below).
- keep the minimum log_size logic and write it directly to the `/etc/rsyslog.conf` file 
- create a new script named something like `ns-log-size` or similar
- the new script does not really need a config inside UCI, but we can use the `/etc/rsyslog.conf` as "database" (the file is preserved between upgrades); the script could have 2 actions:
  - get: that read the current value
  - set: set a new value

I've asked Copilot for a proof of concept (not really tested):
```bash
#!/bin/bash

CONFIG_FILE="/etc/rsyslog.conf"
LOG_ROTATION_LINE=$(grep -E '^\$outchannel log_rotation' "$CONFIG_FILE")

get_log_rotation_size() {
    echo "$LOG_ROTATION_LINE" | awk -F', ' '{print $2}'
}

set_log_rotation_size() {
    local new_size=$1
    if [[ ! $new_size =~ ^[0-9]+$ ]]; then
        echo "Error: Size must be a positive integer."
        exit 1
    fi
    sed -i -E "s/(\$outchannel log_rotation,\/var\/log\/messages, )[0-9]+(, \/usr\/sbin\/rotate-messages)/\1$new_size\2/" "$CONFIG_FILE"
}

case $1 in
    get)
        get_log_rotation_size
        ;;
    set)
        if [ -z "$2" ]; then
            echo "Error: No size provided for set action."
            exit 1
        fi
        set_log_rotation_size "$2"
        ;;
    *)
        echo "Usage: $0 {get|set <size>}"
        exit 1
        ;;
esac
```

The new script can be called from command line or, in the future, from the UI